### DISCUSSION TITLE
Find() / FindAsync()

### DISCUSSION THREAD

**Discussion by DiegoStefanon** [linq2db/linq2db#3381] [General]
Why Linq2Db has no Find() / FindAsync() method like entity fram.?

it would be very useful to have a method that searches for primary keys without the need to insert the field to search for,

many thanks.

**Comment by MaceWindu** [comment]
Well, nobody asked for it. We support Find methods generation by T4, but there is no built-in method for it.

**Reply by DiegoStefanon** [reply]
Ok ;-)
but il would be very useful for develop base data access designed classes because you have not to specify the field name of primary key like in entity framework for example ,very useful really, it infers the primary key so it is very simple, otherwise you have to write Where(f=> f.FIELDNAME == xy)

many thanks

**Comment by DiegoStefanon** [comment]
is possible for future release? like in entity framework ?
a generic dataAccess class in L2db?
Now it is impossible without extra editing...

 public async Task&lt;TbLT&gt; FindAsync(PKT id)
        {
           // return await _genericDbContext.GetTable&lt;TbLT&gt;().Find(id);  
           return await _dbContext.GetTable&lt;TbLT&gt;().FindAsync(id);  
        }

in entity framework you can write:

 return await _dbContext.Set&lt;TbLT&gt;().AsNoTracking().FindAsync(id);

//implementation =>
public virtual ValueTask&lt;TEntity?&gt; FindAsync(params object?[]? keyValues)
public virtual ValueTask&lt;TEntity?&gt; FindAsync(object?[]? keyValues, CancellationToken cancellationToken)


;-)
many thanks

**Reply by MaceWindu** [reply]
available since v5
